site/public/js/gradeable-notebook.js

```diff
@@ -94,43 +129,47 @@ function saveNotebookToLocal() {
  */
 function restoreNotebookFromLocal() {
     if (typeof autosaveEnabled !== 'undefined' && autosaveEnabled) {
-        const state = JSON.parse(localStorage.getItem(notebookAutosaveKey()));
+        const inputs = JSON.parse(localStorage.getItem(notebookAutosaveKey()));
 
-        if (state === null) {
+        if (inputs === null) {
             return;
         }
-
-        // First, we restore multiple choice answers
-        for (const id in state.multiple_choice) {
-            const values = state.multiple_choice[id];
-            // Extract the index from the ID generated from gatherInputAnswersByType()
-            // Index is stored after multiple_choice_, so substring it out
-            const index = id.substring('multiple_choice_'.length);
-            $(`#mc_field_${index} :input`).each((_index, element) => {
-                $(element).prop('checked', values.includes(element.value)).change();
+        //restore multiple choice
+        for (const id in inputs.multiple_choice) {
+            const filename = inputs.multiple_choice[id][0];
+            const answers = inputs.multiple_choice[id][1];
+            $('.multiple_choice').each(function () {
+                $(this).children('fieldset').each(function () {
+                    //match file names
+                    if ($(this).attr('name') === filename) {
+                        $(this).children('label').each(function () {
+                            $(this).children('input').each(function () {
+                                //check proper inputs
+                                for (let i = 0; i < answers.length; i++) {
+                                    if ($(this)[0].defaultValue === answers[i]) {
+                                        $(this).prop('checked', true);
+                                    }
+                                }
+                            });
+                        });
+                    }
+                });
```

Comment:
```suggestion
            $(`fieldset.mc[data-filename="${filename}"] input`).each(function() {
                //check proper inputs
                for (let i = 0; i < answers.length; i++) {
                    if ($(this)[0].defaultValue === answers[i]) {
                        $(this).prop('checked', true);
                    }
                }
            });
```
This can be condensed to remove unnecessary looping. Note that you can use jQuery's [.find()](https://api.jquery.com/find/) method which is similar to .children() except children only searches immediate children while find searches all children no matter how many levels there are. 

This selector will select all `input` elements that are a child (not necessarily direct because we separated by a space, not a `>`) of all `fieldset`s with the class `mc` and the `data-filename` attribute equal to the filename.